flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriterImplTest.java

```diff
@@ -135,145 +144,196 @@ public void testAbortOldAndStartNewCheckpoint() throws Exception {
+                                }
+                                writer.start(
+                                        maxCheckpoints,
+                                        CheckpointOptions.forCheckpointWithDefaultLocation());
+                            }
+                        })
+                .isInstanceOf(IllegalStateException.class);
     }
 
-    @Test(expected = IllegalStateException.class)
-    public void testStartNotOpened() throws Exception {
-        unwrappingError(
-                IllegalStateException.class,
-                () -> {
-                    try (ChannelStateWriterImpl writer =
-                            new ChannelStateWriterImpl(TASK_NAME, 0, getStreamFactoryFactory())) {
-                        callStart(writer);
-                    }
-                });
+    @Test
+    void testStartNotOpened() {
+        assertThatThrownBy(
+                        () ->
+                                unwrappingError(
+                                        IllegalStateException.class,
+                                        () -> {
+                                            try (ChannelStateWriterImpl writer =
+                                                    new ChannelStateWriterImpl(
+                                                            TASK_NAME,
+                                                            0,
+                                                            getStreamFactoryFactory())) {
+                                                callStart(writer);
+                                            }
+                                        }))
+                .isInstanceOf(IllegalStateException.class);
     }
 
-    @Test(expected = IllegalStateException.class)
-    public void testNoStartAfterClose() throws Exception {
-        unwrappingError(
-                IllegalStateException.class,
-                () -> {
-                    ChannelStateWriterImpl writer = openWriter();
-                    writer.close();
-                    writer.start(42, CheckpointOptions.forCheckpointWithDefaultLocation());
-                });
+    @Test
+    void testNoStartAfterClose() {
+        assertThatThrownBy(
+                        () ->
+                                unwrappingError(
+                                        IllegalStateException.class,
+                                        () -> {
+                                            ChannelStateWriterImpl writer = openWriter();
+                                            writer.close();
+                                            writer.start(
+                                                    42,
+                                                    CheckpointOptions
+                                                            .forCheckpointWithDefaultLocation());
+                                        }))
+                .isInstanceOf(IllegalStateException.class);
```

Comment:
```suggestion
        Assertions.assertThatThrownBy(
                        () -> {
                            ChannelStateWriterImpl writer = openWriter();
                            writer.close();
                            writer.start(42, CheckpointOptions.forCheckpointWithDefaultLocation());
                        })
                .hasCauseInstanceOf(IllegalStateException.class);
```
Maybe we can using `hasCauseInstanceOf` directly instead of `unwrappingError`.